### PR TITLE
Fix/#12

### PR DESCRIPTION
**Pull request by FilipDolnik**

### CODE REVIEW

File: Tests/CuckooAPITest.swift [hunk 1]
```diff
@@ -37,49 +22,72 @@ class CuckooAPITest: XCTestCase {
             when(mock.readWriteProperty.set(anyInt())).then {
                 print($0)
             }
+            when(mock.optionalProperty.set(equalTo(nil))).then { _ in
+                when(mock.optionalProperty.get).thenReturn(nil)
+            }
+            when(mock.optionalProperty.set(anyInt())).then { _ in
```
Comment (TadeasKriz) [inline Tests/CuckooAPITest.swift]: We should test if putting in a value will work, like: `mock.optionalProperty.set(10)`

File: Tests/CuckooAPITest.swift [hunk 2]
```diff
@@ -37,49 +22,72 @@ class CuckooAPITest: XCTestCase {
             when(mock.readWriteProperty.set(anyInt())).then {
                 print($0)
             }
+            when(mock.optionalProperty.set(equalTo(nil))).then { _ in
+                when(mock.optionalProperty.get).thenReturn(nil)
```
Comment (TadeasKriz) [inline Tests/CuckooAPITest.swift]: Wait a minute, you cant do this AFAIK. Stubbing outside of the `stub` block is UB.

File: Tests/TestedProtocol.swift [hunk 3]
```diff
@@ -19,9 +21,11 @@ protocol TestedProtocol {
     
     func withThrows() throws
     
-    func withClosure(closure: String -> Int)
+    func withClosure(closure: String -> Int) -> Int
```
Comment (TadeasKriz) [inline Tests/TestedProtocol.swift]: This should be `withClosureReturiningInt` and it should be a new method. Do not change methods already being tested.

File: Tests/CuckooAPITest.swift [hunk 4]
```diff
@@ -88,43 +96,71 @@ class CuckooAPITest: XCTestCase {
...
         verify(mock, never()).withThrows()
-        
+        verify(mock).withClosure(anyClosure())
+        verify(mock).withOptionalClosure("hello", closure: anyClosure())
     }
-        
-}
\ No newline at end of file
+    
+    private enum TestError: ErrorType {
```
Comment (TadeasKriz) [inline Tests/CuckooAPITest.swift]: I'm not too fond of moving this enum outside of the functions as each test can define its own. But for now let's leave it as you changed it.

File: Tests/TestedProtocol.swift (hunk 3 above)
Comment (FilipDolnik) [inline Tests/TestedProtocol.swift]: If I remember correctly there wasn`t any test for withClosure before.

Comment (TadeasKriz) [inline Tests/TestedProtocol.swift]: This diff shows otherwise. Maybe it was the one you commented out, which it should not be.

**Pull request merged**